### DISCUSSION TITLE
compute_dtt RuntimeWarning

### DISCUSSION THREAD

**Discussion by gmsilveira** [ROBelgium/MSNoise#341] [Q&A]
I'm using msnoise to compute dv/v on a single station, considering also the cross-components. 
I'm trying to apply a filter between 0.5 and 1.0 Hz and, for some days, I get the following RuntimeWarning 
**************
msnoise/s06compute_dtt.py:139: RuntimeWarning: invalid value encountered in double_scalars
  wavg = (d * w).sum() / w.sum()
s06compute_dtt.py:141: RuntimeWarning: invalid value encountered in double_scalars
  wstd = np.sqrt(np.sum(w * (d - wavg) ** 2) / ((N - 1) * np.sum(w) / N))

**********************
Can it be related to my filter options including the mwcs length and step?

I've just compute dv/v cross-correlations in a lower frequency band (0.1 - 1.0 Hz) and all went fine.

Many thanks for any suggestion.

Graça Silveira

**Comment by ThomasLecocq** [comment]
@asyates did you have a look at that? replacing the "/" by "np.divide"?

**Reply by asyates** [reply]
Not one I'm so familiar with. This is assuming that the source of error is a division by zero i.e. w.sum() = 0?

@gmsilveira, on the days that have the warning, do the errors in the corresponding MWCS files appear as expected? i.e. non-zero or otherwise. 

That is, for a given day, there is a corresponding .txt file for each station/component pair in the MWCS directory that contains as columns the lag time, delay time measured, error, and coherence. w.sum() is based on the value in the errors column. Wondering if this shows anything unusual (errors much larger than your dtt_maxerr or alternatively non-numeric error values).

**Reply by ThomasLecocq** [reply]
Remember that msnoise2 doesn't make txt files anymore :)

**Reply by asyates** [reply]
Ah, I made the assumption that this was using release version. Is it dev version ?

**Comment by gmsilveira** [comment]
In fact I am using both, as in my new MAC OS with M2 it was easier to install msnoise master.
But those zero divisions happened in version 1.6.?



Professora Coordenadora
Departamento  de Física
Instituto Superior de Engenharia de Lisboa
Rua Conselheiro Emidio Navarro,1
1959-007 Lisbon
Portugal
&
Instituto Dom Luiz IDL
Edif. C8, FCUL, Campo Grande
1749-016 LISBOA
Portugal
Email: ***@***.***



On 2 Jul 2024, at 20:23, Alec Yates ***@***.***> wrote:



Ah, I made the assumption that this was using release version. Is it dev version ?
